cmd/acb/commands/exec/exec.go

```diff
@@ -221,9 +274,21 @@ var Command = cli.Command{
 			credentials = append(credentials, cred)
 		}
 
-		task, err := graph.UnmarshalTaskFromString(ctx, rendered, defaultWorkingDirectory, defaultNetwork, defaultEnvs, credentials, taskName)
-		if err != nil {
-			return err
+		// If work has been done before as a result of preprocessing avoid re
+		// computing the TaskFrom a string.
+		if shouldIncludeAlias {
+			log.Printf("(exec) task completion started")
+			err := task.CompleteTask(ctx, defaultWorkingDirectory, defaultNetwork, defaultEnvs, credentials, taskName)
```

Comment:
Can you rename the method to something else? CompleteTasks suggests that the task will complete/run in this method.